### PR TITLE
Outgoingmessage client v3

### PR DESCRIPTION
**Pull request by lasrinnil**

### CODE REVIEW

File: source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingTests.cs [hunk 1]
```diff
@@ -108,26 +108,32 @@ public async Task Can_peek_oldest_bundle()
     public async Task Can_dequeue_bundle()
     {
         var message = _outgoingMessageDtoBuilder.Build();
-        await EnqueueMessage(message);
-        var peekCommand = new PeekCommand(
+        await EnqueueAndCommitAsync(message);
+        var peekCommand = new PeekRequestDto(
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingTests.cs:112]:
```suggestion
        var peekRequest = new PeekRequestDto(
```

File: source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingTests.cs [hunk 2]
```diff
@@ -108,26 +108,32 @@ public async Task Can_peek_oldest_bundle()
...
             message.ReceiverId,
             MessageCategory.Aggregations,
             message.ReceiverRole,
             DocumentFormat.Xml);
-        var peekResult = await InvokeCommandAsync(peekCommand);
-        var dequeueCommand = new DequeueCommand(
+        var peekResult = await _outgoingMessagesClient.PeekAndCommitAsync(peekCommand, CancellationToken.None);
+        var dequeueCommand = new DequeueRequestDto(
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingTests.cs:118]:
```suggestion
        var dequeueRequest = new DequeueRequestDto(
```

File: source/OutgoingMessages.Interfaces/Models/DequeueRequestDto.cs [hunk 3]
```diff
@@ -15,6 +15,6 @@
 using Energinet.DataHub.EDI.BuildingBlocks.Domain.Actors;
 using Energinet.DataHub.EDI.BuildingBlocks.Domain.Models;
 
-namespace Energinet.DataHub.EDI.OutgoingMessages.Interfaces;
+namespace Energinet.DataHub.EDI.OutgoingMessages.Interfaces.Models;
 
-public record DequeueCommand(string MessageId, MarketRole MarketRole, ActorNumber ActorNumber) : ICommand<DequeCommandResult>;
+public record DequeueRequestDto(string MessageId, MarketRole MarketRole, ActorNumber ActorNumber) : ICommand<DequeueRequestResultDto>;
```
Comment (RasmusGraabaek) [inline source/OutgoingMessages.Interfaces/Models/DequeueRequestDto.cs:20]:
```suggestion
public record DequeueRequestDto(string MessageId, MarketRole MarketRole, ActorNumber ActorNumber) ;
```

**Review by RasmusGraabaek — APPROVED** [review]

File: source/OutgoingMessages.Interfaces/Models/DequeueRequestDto.cs (hunk 3 above)
Comment (lasrinnil) [reply source/OutgoingMessages.Interfaces/Models/DequeueRequestDto.cs:20]: nice thanks!

File: source/IntegrationTests/Application/OutgoingMessages/WhenADequeueIsRequestedTests.cs [hunk 4]
```diff
@@ -70,14 +70,16 @@ public async Task Dequeue_is_Successful()
...
             MessageCategory.Aggregations,
             MarketRole.EnergySupplier,
-            DocumentFormat.Xml));
-
-        var dequeueResult = await InvokeCommandAsync(new DequeueCommand(peekResult.MessageId.GetValueOrDefault().ToString(), MarketRole.EnergySupplier, ActorNumber.Create(SampleData.SenderId)));
+            DocumentFormat.Xml),
+            CancellationToken.None);
 
+        var dequeueResult = await _outgoingMessagesClient.DequeueAndCommitAsync(new DequeueRequestDto(peekResult.MessageId.GetValueOrDefault().ToString(), MarketRole.EnergySupplier, ActorNumber.Create(SampleData.SenderId)), CancellationToken.None);
+        await GetService<ActorMessageQueueContext>().SaveChangesAsync();
```
[the remark this answers is not in the document]
Comment (lasrinnil) [reply source/IntegrationTests/Application/OutgoingMessages/WhenADequeueIsRequestedTests.cs:82]: We shouldn't - have removed the line, it was not needed.

**Review by RasmusGraabaek — APPROVED** [review]

**Pull request merged**